torch/csrc/distributed/rpc/tensorpipe_utils.cpp

```diff
@@ -47,6 +47,12 @@ std::tuple<tensorpipe::Message, TensorpipeWriteBuffers> tensorpipeSerialize(
   tensorpipe::Message tpMessage;
   TensorpipeWriteBuffers buffers;
 
+  // The pickler below might allocate new tensors if RPC arguments contain
+  // Tensor views. Apply stream guard here to include those Tensor allocation
+  // operations to the streams.
+  c10::MultiStreamGuard guard(
+          ctx ? ctx->getReservedStreams() : ArrayRef<Stream>({}));
```

Comment:
I guess this happens within `cloneSparseTensors`? If so, it might be better to move this closer to where we invoke `cloneSparseTensors` and, perhaps, limit this guard to a narrower scope. This would put these two pieces of code, which are related, closer together, and would ensure that anyone who modifies one of them is aware of the other.